oak_functions/loader/src/server.rs

```diff
@@ -384,14 +376,23 @@ impl WasmState {
                 OakStatus::ErrInvalidArgs
             });
 
-        let result = match request {
-            Ok(request) => extension.invoke(self, args, request)?,
-            Err(err) => Err(err),
+        let extension = self.find_extension(handle);
+
+        let result = match (extension, request) {
```

Comment:
If request is an error, why even look up the extension and then check two errors at once? It seems it could be checked in the right order.

Reply:
I think this will be easier if everything before `from_oak_status_result(result)` can be refactored into a function returning `Result<???, OakStatus >` to be able to use the `?` operator.